Add ggdeals to match page

DavidSchor opened the pull request:
As you can see I am not usually a front ender. So it can likely be improved. But this works (however inefficiently).

DavidSchor committed:
Add ggdeals to match page

As you can see I am not usually a front ender. So it can likely be improved. But this works (however inefficiently).

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -146,6 +146,44 @@ function barterReady() {
+                    </a>
+                    <small id="ggdeals_${barterId}_after"></small>
+                </span>`);
+        $(`[id="ggdeals_${barterId}"]`).click(() => {
+            (async() => {
```
Simply `click(async () => {` should work too

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -146,6 +146,44 @@ function barterReady() {
+                    "method": "GET",
+                    "url":    `https://gg.deals/steam/${type}/${id}`,
+                    "onload": (response) => {
+                        let parser = new DOMParser();
+                        let body = parser.parseFromString(response.responseText, "text/html");
+                        let price = $("#game-header-current-prices .price", body).get()
```
jquery doesn't need DOM, u can simply use a string
```suggestion
let price = $("#game-header-current-prices .price", response.responseText).get()
```

DavidSchor replied:
Changing this creates a lot of errors in the console.
"Refused to load the image 'https://barter.vg/images/svgs/icons.svg?v=53e6e40d' because it violates the following Content Security Policy directive: "img-src data: https://bartervg.com https://steamcdn-a.akamaihd.net https://*.steamstatic.com".

I left it as it was.

Revadike replied:
you can ignore that error

DavidSchor replied:
Are you sure? It creates literally hundreds of it. I would rather keep the long form, but you own it, so if you prefer the shorter (and easier to read) but hundreds of warning creating version I will go with that of course.

Revadike replied:
I know why it happens. It's because when jquery parses html string it tends to load all images, which is dumb. Anyway, leave it as it is. It's fine.

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -146,6 +146,44 @@ function barterReady() {
         });
+    
+    // Add GGdeals for match screen
+    $(".matchcol li a").get()
```
```suggestion
    $(".matchcol li a[href*='barter.vg/i/']").get()
```

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -146,6 +146,44 @@ function barterReady() {
+    $(".matchcol li a").get()
+        .forEach((elem) => {
+
+        if (elem.href.includes("barter.vg/bundle/") || elem.href.includes("barter.vg/u/")) {
+            return;
+        }
```
```suggestion
```

DavidSchor committed:
Update Enhanced Barter.user.js

better filtering

Co-authored-by: Revadike <4411977+Revadike@users.noreply.github.com>

DavidSchor committed:
Update Enhanced Barter.user.js

Bad filtering no longer needed

Co-authored-by: Revadike <4411977+Revadike@users.noreply.github.com>

DavidSchor committed:
Implemented suggested changes and fixed multiple calls

I kept the use of dom body in jQuery as I got an error I dont understand about forbidden image access or something in the simpler version. (line 173-175)

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -146,6 +146,41 @@ function barterReady() {
+                    <a style="cursor: pointer;" id="ggdeals_${spanId}">
+                        <img src="https://bartervg.com/imgs/ico/gg.png" width="18" height="18" title="GG.Deals: Click to load price info!">
+                    </a>
+                    <small id="ggdeals_${barterId}_after"></small>
```
```suggestion
                    <small id="ggdeals_${spanId}_after"></small>
```

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -146,6 +146,41 @@ function barterReady() {
             }));
         });
+    
+    let tempI = 0;
```
You sure this is the right solution? Because ideally, we cant to display prices for all the same appids on the page with 1 click, if u know what I mean. It wouldn't be ideal to load the same ggdeals app price more than once on the same page.

DavidSchor committed:
Added user notification after click

Added user notification after click

Revadike commented on Enhanced Barter.user.js:
```diff
@@ -163,6 +163,9 @@ function barterReady() {
                 </span>`);
         $(`[id="ggdeals_${spanId}"]`).click(async () => {
+                // inform user as this can take a bit with calls to barter
+                $(`[id="ggdeals_${barterId}_after"]`).html(` (<a title="GG.Deals current lowest price (Official Stores / Keyshops)">Loading Price please wait</a>)`);
```
```suggestion
                $(`[id="ggdeals_${barterId}_after"] a`).text(`Loading price, please wait`);
```

DavidSchor replied:
I will omit the a as the id should be sufficient and adding the a makes the selector not hit.

DavidSchor committed:
Update Enhanced Barter.user.js

removed un-neaded title on element to make code clearer.
